jax/_src/numpy/fft.py

```diff
@@ -32,6 +33,15 @@
 def _fft_norm(s: Array, func_name: str, norm: str) -> Array:
   if norm == "backward":
     return jnp.array(1)
+
+  # Avoid potential integer overflow
+  if s.dtype.kind in "ui":
+    MAX_LOG = 63 if config.enable_x64.value else 31
+    MAX_LOG *= np.log(2)
+    log_val = reductions.sum(ufuncs.log(s))
+    if log_val >= MAX_LOG:
+      s = s.astype(jnp.float32)
```

Comment:
This looks very expensive to just determine if the conversion to float is required.

A simpler approach would be to always convert all integer inputs to float inputs to make sure that there will be no integer overflow:
```suggestion
    s = s.astype(jnp.float32)
```
or
```suggestion
    s = s.astype(jnp.float64 if jax.config.jax_enable_x64 else jnp.float32)
```